compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensions.cpp

```diff
@@ -916,5 +917,75 @@ void transform_dialect::WorkgroupSwizzleOp::getEffects(
   transform::modifiesPayload(effects);
 }
 
+//===----------------------------------------------------------------------===//
+// TestVectorLayoutAnalysisOp
+//===----------------------------------------------------------------------===//
+
+static void setAnchorOpsFromAttributes(VectorLayoutAnalysis &analysis,
+                                       func::FuncOp funcOp) {
+  for (auto &block : funcOp) {
+    for (auto &op : block) {
+      for (auto attr : op.getAttrs()) {
+        std::string name = attr.getName().str();
+        if (name.find("__vector_layout_test_anchor_operand_") !=
+            std::string::npos) {
+          int operandNum = std::stoi(name.substr(name.find_last_of("_") + 1));
+          analysis.setAnchor(op.getOperand(operandNum), attr.getValue());
+        }
+        if (name.find("__vector_layout_test_anchor_result_") !=
+            std::string::npos) {
+          int resultNum = std::stoi(name.substr(name.find_last_of("_") + 1));
+          analysis.setAnchor(op.getResult(resultNum), attr.getValue());
+        }
+      }
+    }
+  }
+}
+
+static void emitLayoutRemarks(VectorLayoutAnalysis &analysis,
+                              func::FuncOp funcOp) {
+  funcOp.walk([&](Operation *op) {
+    // Do not emit remarks for conflict operations.
+    if (isa<VectorExt::LayoutConflictResolutionOp>(op)) {
+      return;
+    }
+
+    for (OpResult result : op->getOpResults()) {
+      if (auto layout = analysis.getLayout<Attribute>(result)) {
+        // Print layout attr to a string.
+        std::string layoutStr;
+        llvm::raw_string_ostream s(layoutStr);
+        s << layout;
+        // Emit remark.
+        op->emitRemark("layout of result #" +
+                       std::to_string(result.getResultNumber()) + " is " +
```

Comment:
Use a Twine for the former and then you can just plus.

Reply:
or `llvm::formatv`